src/expr.rs

```diff
@@ -1158,6 +1164,30 @@ impl<'a> Rewrite for ControlFlow<'a> {
     }
 }
 
+fn allow_single_line_if(result: &str, block: &ast::Block) -> bool {
+    if result.contains('\n') {
+        return false;
+    }
```

Comment:
`result` here is the condition string, so it's saying if we haven't gone to a new line due to the condition, proceed with checking if we can go for 1 line with the block.

Reply:
I ask because I overlooked this case when adding `single_line_let_else_max_width` and then we needed https://github.com/rust-lang/rustfmt/pull/5902 to correct the issue. I want to make sure that we handle this up front. It would be unfortunate if leading attributes or comments prevented the guard clause from being reformatted.